### PR TITLE
Fixes #1081, Add more profile info on user Profile

### LINKED ISSUE #1081: Add more info on user profile
Something that came up during Hackweek. 
Eg. user's twitter handle.

### PR DESCRIPTION
**Pull request by BelieveC**
Fixes #1081 
Adds info for following social media and social code

- Twitter
- Linkedin
- Github
- Gna
- Gnu
- Savannah
- Google plus
- Diaspora
- Website url

screenshot from 2017-03-09 14-46-17

Working on Test for same.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -8,22 +8,32 @@ def show
...
-    else
-      flash[:error] = "An error prohibited your Profile from being saved: #{@user.errors.full_messages.join('. ')}."
-      render :edit
+    respond_to do |format|
+      if @user.update(user_params)
+        flash[:notice] = 'User was successfully updated.'
+      else
+        flash[:error] = "An error prohibited your Profile from being saved: #{@user.errors.full_messages.join('. ')}."
+      end
+      format.js
```
Comment (ChrisBr) [inline app/controllers/users_controller.rb]: Why do you not render the erros in js?

**Comment by ChrisBr** [comment]
I don't really see the point doing everything with AJAX except making the code more hard to understand IMHO

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -8,22 +8,32 @@ def show
 
   # GET /users/1/edit
   def edit
+    @tab_select = params[:tab]
+    if @tab_select.present?
```
Comment (ChrisBr) [inline app/controllers/users_controller.rb]: Please don't do that with AJAX. Just use the bootstrap tabs to hide and show the different partials.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (ChrisBr) [inline app/controllers/users_controller.rb]: Why is the format.js necessary at all as you didn't change the form to use AJAX?

**Comment by ChrisBr** [comment]
Please also provide screenshots of the tabs and also of the user show page.

**Comment by lagartoflojo** [comment]
Do we really need all of this?
I mean, who uses all these random social media sites?
Why not just add a single field, "Social media profile link", and as help text for that single field, put the Twitter and Facebook examples.
Maybe include "personal website link" as well.
But anything else is complete overkill in my opinion.

👎

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (BelieveC) [inline app/controllers/users_controller.rb]: @ChrisBr Initially even I did the same. I used bootstrap tab to render different partial for sake of simplicity. But the problem that I faced with Bootstrap tabs are as follows:

- It just add `.active` class to a current tab. i.e  Hides the partials of inactive tab
- When I tried submitting the form with this approach. Inactive tab fields acted as hidden fields to the form and also got submitted. 
This is the issue which @differentreality was talking about in previous PR #1180 
>Perhaps. The user, however, will be saving data that are not visible (ie data on the other tab will be saved as well).
 
To solve above issues, Option I had were as follows:
- Implement separate form in a separate tab to detach it from rest of the inactive tab fields. I really didn't like this approach
- Use Ajax to add and remove necessary and unnecessary fields respectively. 

Looking forward to suggestion.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (BelieveC) [inline app/controllers/users_controller.rb]: > Why is the format.js necessary at all as you didn't change the form to use AJAX?
 
I have used it to render different form partial as discusses in above comments.

**Comment by BelieveC** [comment]
Screenshots
- Personal Tab
screenshot from 2017-03-17 09-33-00
- Social Media Tab
screenshot from 2017-03-17 09-33-05
- Social Code Tab
screenshot from 2017-03-17 09-33-08
User show page:
screenshot from 2017-03-17 09-34-36

**Comment by differentreality** [comment]
I am not against giving options (for social media to be added), but I would like to limit the chaos a bit. Does it make sense to put a dropdown with available options of social media? User selects from dropdown, enters url in an input box next to the drop down. 
We only show what the user has.

**Comment by lagartoflojo** [comment]
What would be the drop-down's function?

**Comment by lagartoflojo** [comment]
Also, what's the difference between "social media" and "social code"? Maybe I'm not "social" enough to understand 😝

**Comment by differentreality** [comment]
drop down -> List of social media -> User chooses one -> enters profile link (in the input box) -> then save I guess.

My understanding is that social code is social media where you share your code, eg. github.

**Comment by BelieveC** [comment]
@lagartoflojo 
> Do we really need all of this?
I mean, who uses all these random social media sites?

I agree with your opinion. Actually, these social media sites were already considered in the #1180. This pull request was indeed to implement tab as suggested by @ChrisBr 

> Maybe a second tab with "Contact Information" / "Social Media"?

**Comment by BelieveC** [comment]
@differentreality  and @lagartoflojo 

I think we should consider following social sites:

- Twitter
- Github
- Linkedin( Maybe)

Personally, I would also like to add

- Website Url

What do you think?

**Comment by lagartoflojo** [comment]
I'm not sure then that the drop-down is useful for much other than making people click more.
Why not just a add "social media link" and "personal website link" fields, that's it? Most people have one favorite social media site, and if anything they can link to another one from the "personal website link".
What's the use case of doing anything more?

**Comment by BelieveC** [comment]
@differentreality @lagartoflojo 

I agree with @lagartoflojo 

> just add "social media link" and "personal website link" fields

Finally what should I do?

**Comment by differentreality** [comment]
Closing due to inactivity. We should probably discuss on the issue what it is that we want to do exactly (which fields to add, how to show them, etc)

**Pull request closed without merging**